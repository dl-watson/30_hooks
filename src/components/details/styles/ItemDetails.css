.ItemDetails {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    img {
        height: 20vh;
    }

    .details {
        text-align: left;
    }
}