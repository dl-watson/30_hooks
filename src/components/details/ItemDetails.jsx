import React from "react";
import PropTypes from "prop-types";
import styles from "./styles/ItemDetails.css";

const ItemDetails = ({ character }) => {
  [character] = character;
  return (
    <div className={styles.ItemDetails}>
      {character.name}
      <img src={character.image} />
      <div className={styles.details}>
        <p>Quote: {character.quote}</p>
        <p>Personality: {character.personality}</p>
        <p>Birthday: {character.birthday}</p>
        <p>Species: {character.species}</p>
        <p>Phrase: {character.phrase}</p>
      </div>
    </div>
  );
};

ItemDetails.propTypes = {
  character: PropTypes.array,
};

export default ItemDetails;
