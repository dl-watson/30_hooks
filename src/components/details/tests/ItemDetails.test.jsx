import React from "react";
import { render, cleanup } from "@testing-library/react";
import ItemDetails from "../ItemDetails";

describe("ItemDetails component", () => {
  afterEach(() => cleanup());
  it("renders ItemDetails", () => {
    const { asFragment } = render(<ItemDetails character={[{}]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
