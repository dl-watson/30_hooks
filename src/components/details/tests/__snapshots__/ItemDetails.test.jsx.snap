// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails component renders ItemDetails 1`] = `
<DocumentFragment>
  <div
    class="ItemDetails"
  >
    <img />
    <div
      class="details"
    >
      <p>
        Quote: 
      </p>
      <p>
        Personality: 
      </p>
      <p>
        Birthday: 
      </p>
      <p>
        Species: 
      </p>
      <p>
        Phrase: 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
